File: C Modules/Assignments/Assignment no 7/Question-2/Assignment no 3/Type-3/Question_6.c
#include<stdio.h>
int perfect()
{
    int n,i=1,sum=0;
    int *ptr1=&n;
    printf("Enter the number:\n");
    scanf(" %d",ptr1);
    while (i<=*ptr1/2)
    {
        if (*ptr1%i==0)
        sum=sum+i;
        i++;
    }
        if (sum==*ptr1)
        {
            return 0;
        }
        else {
            return 1;
        }

}
int main()
{
    int result=perfect();
    int *ptr2=&result;
        if (*ptr2==0)
        {
            printf("is a perfect number");
            /* code */
        }
        else {
            printf(" is a not perfect number");
        }

    return 0;
    
}

File: Java Module/Assignments in c to java/Assignment-03/Question_6.java
public class Question_6 {

    public static void main(String[] args) {
            int n=10,i=1,sum=0;
    // printf("Enter the number:\n");
    // scanf(" %d",&n);
    while (i<=n/2)
    {
        if (n%i==0)
        sum=sum+i;
        i++;
    }
        if (sum==n)
        {
            System.out.printf("%d is a perfect number",n);
            /* code */
        }
        else {
            System.out.printf("%d is a not perfect number",n);
        }
    }
}